ldms/src/ldmsd/test/ldmsd_stream_publish.c

```diff
@@ -32,6 +32,7 @@ static struct option long_opts[] = {
 	{0,          0,                 0,  0 }
 };
 
+void usage(int argc, char **argv) __attribute__((noreturn));
```

Comment:
void implies that there is no return value. what is this for?

Reply:
noreturn means the code exits, not that it is void.
this enables correct determination at the higher scope that a code path terminates after a usage call rather than continuing an using uninit variables. the gcc analyzer doesn't dive down into usage and discover the exit() or abort or whatever else.